tests/Integration/Behaviour/Features/Scenario/Product/Combination/Shop/delete_multishop_combination.feature

```diff
@@ -0,0 +1,281 @@
+
+  Scenario: Remove all the combinations for a specific shop and generate the same combinations again for the same shop
+    When I delete following combinations of product product1 from shop "shop1":
+      | id reference   |
+      | product1SWhite |
+      | product1SBlack |
+      | product1SBlue  |
+      | product1MWhite |
+      | product1MBlack |
+      | product1MBlue  |
+    Then product "product1" should have no combinations for shops "shop1"
+    And product "product1" should not have a default combination for shop "shop1"
+    And I generate combinations in shop "shop1" for product product1 using following attributes:
+      | Size  | [S,M]              |
+      | Color | [White,Black,Blue] |
+    Then product "product1" should have the following combinations for shops "shop1":
+      | id reference   | combination name        | reference | attributes           | impact on price | quantity | is default |
+      | product1SWhite | Size - S, Color - White |           | [Size:S,Color:White] | 0               | 0        | true       |
+      | product1SBlack | Size - S, Color - Black |           | [Size:S,Color:Black] | 0               | 0        | false      |
+      | product1SBlue  | Size - S, Color - Blue  |           | [Size:S,Color:Blue]  | 0               | 0        | false      |
+      | product1MWhite | Size - M, Color - White |           | [Size:M,Color:White] | 0               | 0        | false      |
+      | product1MBlack | Size - M, Color - Black |           | [Size:M,Color:Black] | 0               | 0        | false      |
+      | product1MBlue  | Size - M, Color - Blue  |           | [Size:M,Color:Blue]  | 0               | 0        | false      |
+    And product "product1" default combination for shop "shop1" should be "product1SWhite"
+    And product "product1" should have the following combinations for shops "shop2":
+      | id reference        | combination name        | reference | attributes           | impact on price | quantity | is default |
+      | product1SWhiteShop2 | Size - S, Color - White |           | [Size:S,Color:White] | 0               | 0        | true       |
+      | product1SBlackShop2 | Size - S, Color - Black |           | [Size:S,Color:Black] | 0               | 0        | false      |
+      | product1SBlueShop2  | Size - S, Color - Blue  |           | [Size:S,Color:Blue]  | 0               | 0        | false      |
+      | product1MWhiteShop2 | Size - M, Color - White |           | [Size:M,Color:White] | 0               | 0        | false      |
+      | product1MBlackShop2 | Size - M, Color - Black |           | [Size:M,Color:Black] | 0               | 0        | false      |
+      | product1MBlueShop2  | Size - M, Color - Blue  |           | [Size:M,Color:Blue]  | 0               | 0        | false      |
+    And product "product1" default combination for shop "shop2" should be "product1SWhiteShop2"
+    And following combination ids should match:
```

Comment:
If you implement my earlier suggestion about `combination id` / `id reference` I think we don't need this technical step anymore